### ISSUE TITLE
Compilation Errors

### ISSUE THREAD

**Issue by ghost** [fw4spl-org/fw4spl-deps#10]
I am using Windows 10 and I have been following the instructions on  [http://fw4spl.readthedocs.io/en/dev/Installation/src/WindowsInstall.html](url). In the `cmake-gui`, I did not use the generator `NMake Makefiles JOM` but instead I used `Visual Studio 15 2017 Win64`. The generation was successful. Now I used `Microsoft Visual Studio 2017` to compile the dependencies by using the `BinPkgs` solution file but I am getting these errors below.

I also have log files: [logfiles.zip](https://github.com/fw4spl-org/fw4spl-deps/files/875374/logfiles.zip)

![capture](https://cloud.githubusercontent.com/assets/26401416/24401856/09765dc6-13f1-11e7-95a3-72c1cb4197bc.JPG)

By the way, I have read in [https://github.com/fw4spl-org/fw4spl-deps/issues/9](url) that you mentioned that a precompiled binary will be released for Windows, are there any available?

**Comment by greenjava** [comment]
Hi,
Thank you for your interest with fw4spl and submitting your issue.
Visual Studio 2017 in not yet supported to build fw4spl and fw4spl-deps.

Currently only Visual 2015 (as mentioned in the [doc](http://fw4spl.readthedocs.io/en/dev/Installation/src/WindowsInstall.html#prerequisites-for-windows-users)) is officially supported.

Precompiled  dependencies for Windows/Linux/macOS and Visual 2017 support is planned in the coming months.

Sorry for the inconvenience

**Comment by ghost** [comment]
Hi, this time I used the `Visual Studio 14 2015 Win64` generator in the `cmake-gui` and the generation was successful. Now I used `Microsoft Visual Studio 2015` to compile the dependencies by using the `BinPkgs` solution file and I am getting some errors. Can you help me out?

I also have log files: [Logfiles.zip](https://github.com/fw4spl-org/fw4spl-deps/files/877627/Logfiles.zip)

![capture](https://cloud.githubusercontent.com/assets/26401416/24435745/18e52ed2-1472-11e7-89ce-fa4bf3aad4b2.JPG)

If you do not have precompiled binaries for Visual 2017, do you have any for 2015?

**Comment by greenjava** [comment]
Hi,
OK, I recommend using the generator "**_NMake Makefiles JOM_**" to compile the dependencies (it's faster and it's the method we use here). I note the issue with this generator, thank you for your comment.
I'll try to upload MSVC2015 binaries today.

**Comment by ghost** [comment]
Hi, I did not use `Jom` since there were again errors occurring during the source code compilation which I could not solve.  `Jom` also provides precompiled binaries but their link is broken (even the archives for the older versions). I had no choice but to use the Visual Studio generator.

Thanks, I appreciate your help and I'll be looking forward to the binaries.

**Comment by greenjava** [comment]
hum, strange, I have no problem to download jom (did you look here: https://wiki.qt.io/Jom ?)

The compilation with _Visual Studio generator_ (compared to the _Makefile generator_) is different for some project, _libarchive_ for example adds specific flags _/weXXXX_ for debug build :disappointed:. It requires to patch many lib.

I'll run the build to provide a clean precompiled version (we'll try to automate this for future versions :wink:).

**Comment by ghost** [comment]
Yes, this is the page that is redirected when I try to download Jom. I also tried emailing the server administrator but didn't get any response from them. Could you possibly upload the file for me here? Maybe if I can install Jom, maybe creating the binary would be easier for you.

![capture1](https://cloud.githubusercontent.com/assets/26401416/24448089/f580e620-14ad-11e7-9cba-659fead4642d.JPG)

**Comment by greenjava** [comment]
Ok, the download mirror site proposed is down (_jaist.ac.jp_), embarrassing...
You can force the download on another mirror here:
Https://download.qt.io/official_releases/jom/jom_1_1_2.zip.mirrorlist
(http://master.qt.io/official_releases/jom/jom_1_1_2.zip seems ok)

**Comment by ghost** [comment]
Thanks I'll try it out. Please let me know when the binaries are ready!

**Comment by greenjava** [comment]
I've uploaded a debug and release version (tag 11.0.4) of the dependencies (built with Visual 2015 in x64):
- [fw4spl-deps-debug-11.0.4_20170329_Win64_MSVC2015.zip](https://owncloud.u-strasbg.fr/index.php/s/YMgWmJsgmaI2z8X/download)
- [fw4spl-deps-release-11.0.4_20170329_Win64_MSVC2015.zip](https://owncloud.u-strasbg.fr/index.php/s/dyLSOzAgo0Iwh3A/download)

It's not yet "official" versions (kind of _beta version_), the process need to be automated in build farms.

**Comment by ghost** [comment]
Thanks for your help!

Sorry to bother you :( but I'm also having trouble building `fw4spl` in the `cmake-gui` with `fw4spl-ar` and `fw4spl-ogre` as the `ADDITIONAL_PROJECTS`. What could be the problem? Would it be too much to ask for a precompiled binary for `fw4spl` also? Thanks.

![capture](https://cloud.githubusercontent.com/assets/26401416/24497856/c211d5e0-1577-11e7-9e7c-a397d2e23c3b.JPG)

![1](https://cloud.githubusercontent.com/assets/26401416/24497965/14df32d6-1578-11e7-9b1a-2d4532e8d149.JPG)

![2](https://cloud.githubusercontent.com/assets/26401416/24498031/40b97880-1578-11e7-81d6-9cd99146380c.JPG)

**Comment by greenjava** [comment]
Hi,
The problem here is in **_ADDITIONAL_PROJECTS_** value,
you must write CMake path in _**unix-style**_, with "**/**":
`C:/FW4SPL/fw4spl/Dev/Src/fw4spl-ar;C:/FW4SPL/fw4spl/Dev/Src/fw4spl-ogre`

Precompiled binary version for fw4spl is planned (it's the next cool feature), but this requires a lot of work for everything work properly (due to the particular 'bundle' architecture of fw4spl).
